### ISSUE TITLE
Premium block flows: update nudge styles

### ISSUE THREAD

**Issue by simison** [Automattic/jetpack#12946]
@scruffian's design for the premium blocks:

<img width="566" alt="image" src="https://user-images.githubusercontent.com/87168/60615490-e8c0f500-9dd7-11e9-9d0f-a647c283dad7.png">

Since the nudge is in the block editor, safe to assume we can follow Gutenberg UI styles instead of Calypso?

Alternatively, it could be seen as a piece of separate over the editor and use [Calypso banner component](https://github.com/Automattic/wp-calypso/tree/ad190268ff0c526773ee0b7c5d3a6805b861d6b3/client/components/banner) when in .com and [Jetpack banner component](https://github.com/Automattic/jetpack/tree/bf90343998e5eaf0eb832e995def7d9caa1bce0c/_inc/client/components/banner) when shipping in Jetpack.

Even if that would be a goal eventually, we should stick to core styles for now to keep things technically simple, so:

- Block editor's font family and font sizes
- Core button component
- No theme inherited fonts or styles
- Padding/whitespace values from core

Since core doesn't publish block editor's [basic styles](https://github.com/WordPress/gutenberg/blob/9052d4c26fff89907d7e7abec745a3d0079dd178/assets/stylesheets/_variables.scss) to NPM, we've been [copying them over to Jetpack](https://github.com/Automattic/jetpack/blob/67ce7185ed48f520a93c9e3bf8b498a47e4007bd/extensions/shared/styles/gutenberg-variables.scss).

**Comment by simison** [comment]
@mtias sounds good?

@ockham so this is more like 1st iteration, feel free to expand here how 2nd iteration could look like with properly shared components.

**Comment by mtias** [comment]
We should follow the style used for invalid block / missing block in core. And it might be good to expose this as a "card" UI component in `wordpress/packages` if it's going to be useful and more consistent for others. (This is not blocking.)

**Comment by simison** [comment]
>We should follow the style used for invalid block / missing block in core. 

#### Current upgrade nudge (selected + unselected):

<img width="795" alt="Screenshot 2019-07-04 at 11 48 36" src="https://user-images.githubusercontent.com/87168/60652976-cec3f880-9e51-11e9-95ad-265fef280a0e.png">
<img width="774" alt="Screenshot 2019-07-04 at 11 48 19" src="https://user-images.githubusercontent.com/87168/60652978-cf5c8f00-9e51-11e9-86e2-e461ed3d4ed5.png">

#### Invalid block (selected + unselected):

<img width="492" alt="Screenshot 2019-07-04 at 11 50 13" src="https://user-images.githubusercontent.com/87168/60653057-f61ac580-9e51-11e9-8469-5f4ae97257d8.png">
<img width="491" alt="Screenshot 2019-07-04 at 11 50 16" src="https://user-images.githubusercontent.com/87168/60653056-f61ac580-9e51-11e9-860c-75be987b3b8e.png">

#### Re-usable block (selected + unselected):

<img width="502" alt="Screenshot 2019-07-04 at 11 43 02" src="https://user-images.githubusercontent.com/87168/60653136-1f3b5600-9e52-11e9-8cf7-5eb4af61893d.png">
<img width="495" alt="Screenshot 2019-07-04 at 11 52 11" src="https://user-images.githubusercontent.com/87168/60653178-32e6bc80-9e52-11e9-9a68-5dbff4c7e2e4.png">

#### Missing block (selected + unselected):
<img width="498" alt="Screenshot 2019-07-04 at 11 58 53" src="https://user-images.githubusercontent.com/87168/60653670-257e0200-9e53-11e9-9a03-dd03c150c046.png">
<img width="504" alt="Screenshot 2019-07-04 at 11 58 56" src="https://user-images.githubusercontent.com/87168/60653669-257e0200-9e53-11e9-878d-43a740efa40a.png">


### Looking at these I'd conclude that:

- Since the customer can interact with unavailable premium block, let's remove the white hue covering the block. So have it more similar to re-usable blocks and invalidated blocks. Hue is not present in the missing block, which is semantically more similar to premium block nudge but you cannot interact with semantic blocks so that the hue makes more sense there. Without the hue, customers get a better sense how it'll look at the site and there won't be color-contrast accessibility issues while they try to edit the block.

- Add different border colors while the block is selected. Look for `.block-editor-warning` and `.is-selected` selectors in the core.

- Probably doesn't make sense to re-use `.block-editor-warning` styles as-is from the core, since the upgrade nudge should still resemble enough other upgrade nudges in Calypso so that it stays more-or-less consistent.

- Note how the block warning banner in core stretches across block container, not only across the block itself. Not a blocker, but we could do the same if it's not hacky.
    <img width="243" alt="Screenshot 2019-07-04 at 11 53 14" src="https://user-images.githubusercontent.com/87168/60654234-44c95f00-9e54-11e9-9497-7962048e7adc.png">

- Let's use matching whitespace from `.block-editor-warning`: `padding: 10px 14px;`

- Keep the button at the right side but make upgrade button center-aligned, similar to [action card](https://wpcalypso.wordpress.com/devdocs/design/action-card) in Calypso or "invalid block" warning when it's on wider screen:
    <img width="790" alt="Screenshot 2019-07-04 at 12 18 20" src="https://user-images.githubusercontent.com/87168/60655126-e7361200-9e55-11e9-8b8c-b6036ca13018.png">
   On narrower screens the button should float under the text left-aligned.

- We talked briefly with @ockham about making the whole banner a button, like they're often in .com:
    ![image](https://user-images.githubusercontent.com/87168/60654569-e6e94700-9e54-11e9-8cb3-afed473d0673.png)

    I think we can keep the CTA button instead because of "click to select the block" nature of this UI. Fewer chances for accidental clicks.

- Should the button become primary-blue when the block is selected? Core itself uses lots of primary buttons but I'd rather not add more to that noise.

- Let's use [block editor's default fonts](https://github.com/WordPress/gutenberg/blob/9052d4c26fff89907d7e7abec745a3d0079dd178/assets/stylesheets/_variables.scss#L6).

- Ensure all this works nicely with RTL styles

**Comment by ockham** [comment]
> Should the button become primary-blue when the block is selected? Core itself uses lots of primary buttons but I'd rather not add more to that noise.

Do we ever change button relevance contextually from default to primary? Seems like a whole new UX pattern :thinking: 

(Generally, we were going for default rather than primary, as per paAmJe-ll-p2 #comment-741)

**Comment by ockham** [comment]
Let's also discuss copy here:

@scruffian's mockups (as seen in this PR's description) orginally had:

> Premium and Business Plans have access to this block.
> The block will be hidden to your visitors until you upgrade.

Since the 'Upgrade' button directly links to Checkout of the Premium plan, we shouldn't mention the Business Plan IMO. (This also makes it easier to keep the upgrade nudge more generic).

The current copy, which I essentially lifted from some P2 post, isn't stellar however:

> Upgrade to Premium
> To make this block visible on your site

(The plan name -- 'Premium'  in this case -- is obtained from the endpoint.)

Maybe at least use the second line from Ben's mockup? (It should be 'hidden from', not 'hidden to' though, right?) And maybe change the first line to something like:

> You need a Premium Plan (or higher) to make this block available.
> The block will be hidden from your visitors until you upgrade.

**Comment by simison** [comment]
@Automattic/editorial could you help us out here? :-) See above comment.

TL;DR: Upgrade nudge is shown in the block editor when customer's plan (free, personal) doesn't have access the block they've just inserted in the editor. "Upgrade" brings to checkout with plan pre-selected and then drops them right back to the editor after purchase.

**Comment by ockham** [comment]
Another question: Which icon do we wanna use? The mockup uses the pencil (premium plan?), whereas the code currently uses the star (generic upgrade?)

**Comment by simison** [comment]
I would expect to see the pen icon for consistency with .com, but there are some more generic upgrade nudges in Calypso, too:

<img width="489" alt="image" src="https://user-images.githubusercontent.com/87168/60711344-66dae400-9f1d-11e9-9be6-a8a6254351d9.png">

via https://wordpress.com/marketing

Let's keep the start for 1st iteration to keep things simple. We can mention it in call-for-testing to get feedback.

**Comment by mtias** [comment]
Yes, the star is fine for now.

**Comment by kristastevens** [comment]
Hi @simison, thanks for the ping! Here are some suggestions: 

You need a Premium Plan to unlock this block. 
It will be hidden from site visitors until you upgrade.

This block is available under the Premium Plan.
It will be hidden from site visitors until you upgrade.

This block will be hidden to site visitors.
Upgrade to a Premium Plan to unlock this block.

**Comment by drw158** [comment]
> Should the button become primary-blue when the block is selected? Core itself uses lots of primary buttons but I'd rather not add more to that noise.

It should use the default button unless it's the primary action in the view. In that case, it would be primary.
